src/sed.js

```diff
@@ -28,13 +28,21 @@ function _sed(options, regex, replacement, file) {
   else
     common.error('invalid replacement string');
 
+  // Convert all search strings to RegExp
+  if (typeof regex === 'string')
+    regex = RegExp(regex);
+
   if (!file)
     common.error('no file given');
 
   if (!fs.existsSync(file))
     common.error('no such file or directory: ' + file);
 
-  var result = fs.readFileSync(file, 'utf8').replace(regex, replacement);
+  var result_list = fs.readFileSync(file, 'utf8').split('\n').map(function (line) {
+    return line.replace(regex, replacement);
+  });
```

Comment:
I originally wrote it like that. I switched it because I thought we might later want to modify sed to return a list (similar what ls() does) instead of return a multi line string. I can revert this back though